prog/validation.go

```diff
@@ -20,13 +20,13 @@ func init() {
 
 func (p *Prog) debugValidate() {
 	if debug {
-		if err := p.validate(); err != nil {
+		if err := p.Validate(); err != nil {
 			panic(err)
 		}
 	}
 }
 
-func (p *Prog) validate() error {
+func (p *Prog) Validate() error {
```

Comment:
I don't like this. We managed for 10 years w/o exporting this method to other package.
If some operation does not trivial program manipulation and requires access to validation it should be places in the prog package, rather than in some leaf tool.